CSVEnumerable/ResetEnumerator.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+
+namespace CSVEnumerable
+{
+    public class ResetEnumerator<T>: IEnumerator<T>
```

Comment:
Let's call it `CsvEnumerator`. `Reset` word is redundant here because `IEnumerator` in .NET has the following members:
```
    public interface IEnumerator
    {
        bool MoveNext();

        object Current
        {
            get;
        }

        void Reset();
    }
```

Reply:
Another solution - implement the interface `IEnumerator<T>` directly in the class `CsvEnumarable`
```public class CsvEnumerable<T> : IEnumerable<T>, IEnumerator<T> { }```